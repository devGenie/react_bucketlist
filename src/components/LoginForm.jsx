import React from 'react';

class LoginForm extends React.Component {
	constructor(props){
		super(props);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleSubmit(event){
		event.preventDefault();
	}
	render(){
		return (<form onSubmit={this.handleSubmit}>
					<input type="text" id="login_username"/>
					<input type="password" id="login_password"/>
					<button type="submit" id="loginBtn">Login</button>
				</form>)
	}
}

export default LoginForm;