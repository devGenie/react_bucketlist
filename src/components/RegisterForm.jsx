import React from 'react';
import FieldWrapper from './FieldWrapper';

class RegisterForm extends React.Component{
	constructor(props){
		super(props);

		this.state={
			'email':'',
			'first_name':'',
			'last_name':'',
			'password1':'',
			'password2':''
		}
		this.handleSubmit = this.handleSubmit.bind(this);
		this.handleChange = this.handleChange.bind(this);
	}

	handleSubmit(event){
		event.preventDefault();
	}

	handleChange(event){
		const target = event.target;
		const value = target.value;
		const name = target.name;

		this.setState({
			[name]:value
		})
	}
	render(){
			return(
					<form className="col s12 translucent" onSubmit={this.handleSubmit}>
						<FieldWrapper Label='Email'>
							<input type="text" name="email" onChange={this.handleChange} />
						</FieldWrapper>

						<FieldWrapper Label='First Name'>
							<input type="text" name="first_name" onChange={this.handleChange}/>
						</FieldWrapper>

						<FieldWrapper Label='Last Name'>
							<input type="text" name="last_name" onChange={this.handleChange}/>
						</FieldWrapper>

						<FieldWrapper Label='Password'>
							<input type="password" name="password1" onChange={this.handleChange}/>
						</FieldWrapper>

						<FieldWrapper Label='Repeat password'>
							<input type="password" name="password2" onChange={this.handleChange}/>
						</FieldWrapper>

						<FieldWrapper extraz="center">
							<button type="submit" name="submit" className='btn'>Register</button>
						</FieldWrapper>
					</form>
			)}
	}

export default RegisterForm;