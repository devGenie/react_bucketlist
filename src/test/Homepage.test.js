import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import {expect} from 'chai';
import {stub} from 'sinon';
import 'should';
import 'should-enzyme';

import LoginForm from '../components/LoginForm';
import RegisterForm from '../components/RegisterForm';

describe('<LoginForm/>',()=>{
	it('should have two input fields',()=>{
		const wrapper = shallow(<LoginForm/>);
		expect(wrapper.find('input')).to.have.length(2);
	});

	it('should have one username input field',()=>{
		const wrapper = shallow(<LoginForm/>);
		expect(wrapper.find('#login_username').length).equal(1);
	});

	it('should have one password input field',()=>{
		const wrapper = shallow(<LoginForm/>);
		expect(wrapper.find('#login_password')).to.have.length(1);
	});

	it('should have correct input field types',()=>{
		const wrapper = shallow(<LoginForm/>);
		wrapper.find("#login_username").should.have.attr('type','email');
		wrapper.find("#login_password").should.have.attr('type','password')
	})

	it('should have values credentials entered correctly',()=>{
		const credentials={"username":"devgenie","password":"genieinthehouse"};
		const wrapper =shallow(<LoginForm/>);
		expect(wrapper.find('#login_password')).to.have.length(1);
		expect(wrapper.find('#login_username')).to.have.length(1);

		const email=wrapper.find("#login_email");
		email.value=credentials.username;
		expect(email.value).to.equal('devgenie');

		const password= wrapper.find("#login_password");
		password.value = credentials.password;
		expect(password.value).to.equal('genieinthehouse');
	});

	it('has a submit button',()=>{
		const wrapper=shallow(<LoginForm/>);
		var loginBtn=wrapper.find("button #loginBtn");
		expect(loginBtn).to.have.length(1);

		loginBtn.should.have.attr('type','submit');

	});

	it('can be submited',()=>{
		const doSubmit = stub(LoginForm.prototype,'handleSubmit').returns(true);
		const loginForm = shallow(<LoginForm/>);
		loginForm.simulate('submit',{
  				preventDefault: () => {cosole.log("yes")}
		});
		expect(doSubmit.called).to.be.true;
		doSubmit.restore();
	});
});

describe('<RegisterForm', () =>{
	it('should have four input fields', ()=>{
		const wrapper=shallow(<RegisterForm/>);
		expect(wrapper.find('input')).to.have.length(4);
	});

	it('should have a password field', ()=>{
		const wrapper = shallow(<RegisterForm/>);
		expect(wrapper.find({type:'password'})).to.have.length(1);
	});

	it('should have two text input fields',()=>{
		const wrapper = shallow(<RegisterForm/>);
		expect(wrapper.find({type:'text'})).to.have.length(2);
	});

	it('should have an email input fields',()=>{
		const wrapper = shallow(<RegisterForm/>);
		expect(wrapper.find({type:'email'})).to.have.length(1);
	});

	it('should be able to submit',()=>{
		const register = stub(RegisterForm.prototype,'handleSubmit').returns(true);
		const registrationForm = shallow(<RegisterForm/>);

		registrationForm.simulate('submit',{
			preventDefault: () => {}
		});
		expect(register.called).to.be.true;
		register.restore();
	})
})