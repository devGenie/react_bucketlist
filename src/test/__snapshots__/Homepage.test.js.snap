// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen> should render 1`] = `
<HomeScreen>
  <div
    className="genie-carousel"
  >
    <div
      className="row"
      id="forms"
    >
      <div
        className="flipper"
      >
        <div
          id="loginForm_area"
        >
          <LoginForm>
            <form
              className="translucent col s12"
              onSubmit={[Function]}
            >
              <Loading
                status="hide"
              >
                <div
                  className="progress hide"
                >
                  <div
                    className="indeterminate"
                  />
                </div>
              </Loading>
              <div
                className="innerForm col s12"
              >
                <h5
                  className="col s12 center"
                >
                  Login
                </h5>
                <FieldWrapper
                  Label="Email"
                >
                  <div
                    className="input-field  "
                  >
                    <input
                      className="validate"
                      id="login_username"
                      name="email"
                      onChange={[Function]}
                      type="email"
                    />
                    <label>
                      Email
                    </label>
                  </div>
                </FieldWrapper>
                <FieldWrapper
                  Label="Password"
                >
                  <div
                    className="input-field  "
                  >
                    <input
                      aria-required="true"
                      id="login_password"
                      name="password"
                      onChange={[Function]}
                      required=""
                      type="password"
                    />
                    <label>
                      Password
                    </label>
                  </div>
                </FieldWrapper>
                <FieldWrapper
                  extraz="center"
                >
                  <div
                    className="input-field center"
                  >
                    <button
                      className="btn "
                      id="loginBtn"
                      type="submit"
                    >
                      Login
                    </button>
                    <label />
                  </div>
                </FieldWrapper>
              </div>
            </form>
          </LoginForm>
          <Note>
            <p
              className="note translucent center white col s12"
            >
              Don't have an account?
            </p>
          </Note>
          <Fliper>
            <div
              className="col s12 center"
            >
              <button
                className="btn flip_me"
              >
                Register
              </button>
            </div>
          </Fliper>
        </div>
        <div
          id="registerForm_area"
        >
          <RegisterForm>
            <form
              className="col s12 translucent"
              onSubmit={[Function]}
            >
              <Loading
                status="hide"
              >
                <div
                  className="progress hide"
                >
                  <div
                    className="indeterminate"
                  />
                </div>
              </Loading>
              <div
                className="innerForm col s12"
              >
                <h5
                  className="col s12 center"
                >
                  Register
                </h5>
                <FieldWrapper
                  Label="Email"
                >
                  <div
                    className="input-field  "
                  >
                    <input
                      className="validate"
                      name="email"
                      onChange={[Function]}
                      type="email"
                    />
                    <label>
                      Email
                    </label>
                  </div>
                </FieldWrapper>
                <FieldWrapper
                  Label="First Name"
                >
                  <div
                    className="input-field  "
                  >
                    <input
                      className="validate"
                      name="first_name"
                      onChange={[Function]}
                      type="text"
                    />
                    <label>
                      First Name
                    </label>
                  </div>
                </FieldWrapper>
                <FieldWrapper
                  Label="Last Name"
                >
                  <div
                    className="input-field  "
                  >
                    <input
                      className="validate"
                      name="last_name"
                      onChange={[Function]}
                      type="text"
                    />
                    <label>
                      Last Name
                    </label>
                  </div>
                </FieldWrapper>
                <FieldWrapper
                  Label="Password"
                >
                  <div
                    className="input-field  "
                  >
                    <input
                      className="validate"
                      name="password"
                      onChange={[Function]}
                      type="password"
                    />
                    <label>
                      Password
                    </label>
                  </div>
                </FieldWrapper>
                <FieldWrapper
                  extraz="center"
                >
                  <div
                    className="input-field center"
                  >
                    <button
                      className="btn "
                      name="submit"
                      type="submit"
                    >
                      Register
                    </button>
                    <label />
                  </div>
                </FieldWrapper>
              </div>
            </form>
          </RegisterForm>
          <Note>
            <p
              className="note translucent center white col s12"
            >
              Already have an account?
            </p>
          </Note>
          <Fliper>
            <div
              className="col s12 center"
            >
              <button
                className="btn flip_me"
              >
                Login
              </button>
            </div>
          </Fliper>
        </div>
      </div>
    </div>
  </div>
</HomeScreen>
`;
